weather.py

```diff
@@ -17,6 +17,8 @@ def format_temperature(temp):
 
 
 def convert_date(iso_string):
+    date = datetime.fromisoformat(iso_string)
+    return(date.strftime("%A %d %B %Y"))
     """Converts and ISO formatted date into a human readable format.
 
     Args:
```

Comment:
Minor comment: In functions, the comments enclosed in `""" """` are by convention kept at the top right after defining the function, and called `docstrings`. `docstrings` are used to inform other programmers what the function does, and sometimes tools like VS code extensions are able to automatically detect them to help with debugging later on.